### ISSUE TITLE
Explore Alternative Models and Parameter Tuning

### ISSUE THREAD

**Issue by rodelBeronilla** [rodelBeronilla/NBA-Analysis#3]
2. Explore Alternative Models and Parameter Tuning
Experiment with different models and tune their parameters. Scikit-learn offers tools like GridSearchCV for parameter tuning.
python
Copy code
from sklearn.model_selection import GridSearchCV
from sklearn.ensemble import RandomForestClassifier

# Example model: Random Forest
model = RandomForestClassifier(random_state=42)

# Parameters to tune
param_grid = {
    'n_estimators': [100, 200, 300],
    'max_depth': [None, 10, 20, 30],
}

# Setup GridSearchCV
grid_search = GridSearchCV(estimator=model, param_grid=param_grid, cv=5, scoring='accuracy', n_jobs=-1)

# Assuming X_train_scaled and y_train are already defined
grid_search.fit(X_train_scaled, y_train)

# Best model
best_model = grid_search.best_estimator_

**Comment by rodelBeronilla** [comment]
Best model parameters: {'estimator__bootstrap': True, 'estimator__ccp_alpha': 0.0, 'estimator__class_weight': None, 'estimator__criterion': 'gini', 'estimator__max_depth': None, 'estimator__max_features': 'sqrt', 'estimator__max_leaf_nodes': None, 'estimator__max_samples': None, 'estimator__min_impurity_decrease': 0.0, 'estimator__min_samples_leaf': 1, 'estimator__min_samples_split': 2, 'estimator__min_weight_fraction_leaf': 0.0, 'estimator__monotonic_cst': None, 'estimator__n_estimators': 100, 'estimator__n_jobs': None, 'estimator__oob_score': False, 'estimator__random_state': 42, 'estimator__verbose': 0, 'estimator__warm_start': False, 'estimator': RandomForestClassifier(random_state=42), 'n_jobs': -1}
